src/Scope.php

```diff
@@ -25,11 +25,15 @@ class Scope
 
     /**
      * @param Scope $scope
+     * @param string $key - an optional key. defaults to the scope's object hash
      */
-    public function peridotAddChildScope(Scope $scope)
+    public function peridotAddChildScope(Scope $scope, $key = "")
     {
         $scope->peridotSetParentScope($this);
-        $this->peridotGetChildScopes()->attach($scope);
+        if (empty($key)) {
+            $key = spl_object_hash($scope);
+        }
+        $this->peridotChildScopes[$key] = $scope;
```

Comment:
Yeah - and the API provides means to remove an object you didn't supply a key for:

```php
$children = $scope->peridotGetChildScopes();
$child = array_reduce($children, 'meetsMyCriteriaFunc');
$scope->peridotRemoveChildScope(spl_object_hash($child));
```

I imagine this case would be infrequent - but definitely possible